src/cim/components/Hypershift/HostedClusterWizard/NetworkStep/NetworkForm.tsx

```diff
@@ -17,26 +19,30 @@ import {
   SelectField,
   UploadSSH,
 } from '../../../../../common';
-import { useFormikContext } from 'formik';
 import { useTemptiflySync } from '../../hooks/useTemptiflySync';
 import { useTranslation } from '../../../../../common/hooks/use-translation-wrapper';
+import { AgentK8sResource } from '../../../../types';
 
 import './NetworkForm.css';
 
 const NetworkForm: React.FC<NetworkFormProps> = ({ agents, onValuesChanged }) => {
   const { values, setFieldValue } = useFormikContext<NetworkFormValues>();
   useTemptiflySync({ values, onValuesChanged });
 
-  const allIps = flattenDeep<string>(
-    agents
-      .filter((a) => !!a.status?.inventory.interfaces)
-      .map((a) =>
-        (a.status?.inventory.interfaces as Interface[]).map(
-          // eslint-disable-next-line
-          (i) => (i as any).ipV4Addresses as string[],
-        ),
-      ),
+  const allAgents: AgentK8sResource[] = agents.filter((a) => !!a.status?.inventory.interfaces);
+  const allInterfacesDeep: (Interface[] | undefined)[] = allAgents.map(
+    (a) => a.status?.inventory?.interfaces,
   );
+  const allInterfaces: Interface[] = flattenDeep(allInterfacesDeep).filter(Boolean) as Interface[];
+  const allIpsRaw: string[][] = allInterfaces.map((i: Interface): string[] => {
+    // @ts-ignore
+    const ipV4Addresses: string[] = (i.ipV4Addresses as string[]) || ([] as string[]);
+    // @ts-ignore
+    const ipV6Addresses: string[] = (i.ipV6Addresses as string[]) || ([] as string[]);
```

Comment:
we already have a similar conversion in one other place (`toAssisted.ts`). Maybe its time to fix the AgentK8sResource status.inventory type ? The current state is very error prone

In `src/cim/types/k8s/agent.ts`
```ts
type AgentInventory = Inventory & {
  interfaces?: CIMInterface[];
};
```
AgentK8sResource.status.inventory: AgentInventory

It would also allow us to remove the ts-ignore comments

Reply:
I agree, I will push a follow-up with that, we want release asap